Compile faild with type error

# Bug Report

I want build my codes, but faild building my code.

### 💻 Code
```sh

D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/Lazy/flat.d.ts (2,15): ',' expected.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/Lazy/flatMap.d.ts (3,10): Duplicate identifier 'type'.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/Lazy/flatMap.d.ts (3,15): ',' expected.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/Lazy/flatMap.d.ts (3,25): Duplicate identifier 'type'.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/Lazy/flatMap.d.ts (3,30): ',' expected.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts (4,10): Duplicate identifier 'type'.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts (4,10): Cannot find name 'type'.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts (4,15): ',' expected.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts (4,25): Duplicate identifier 'type'.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts (4,25): Cannot find name 'type'.
D:/coding/bamboo/test-server/node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts (4,30): ',' expected.
```

```
yarn typeorm migration

node_modules/@fxts/core/dist/types/Lazy/flat.d.ts:2:15 - error TS1005: ',' expected.

2 import { type DeepFlat, DeepFlatSync } from "../types/DeepFlat";
                ~~~~~~~~

node_modules/@fxts/core/dist/types/Lazy/flatMap.d.ts:3:15 - error TS1005: ',' expected.

3 import { type DeepFlat, type DeepFlatSync } from "../types/DeepFlat";
                ~~~~~~~~

node_modules/@fxts/core/dist/types/Lazy/flatMap.d.ts:3:30 - error TS1005: ',' expected.

3 import { type DeepFlat, type DeepFlatSync } from "../types/DeepFlat";
                               ~~~~~~~~~~~~

node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts:4:15 - error TS1005: ',' expected.

4 export { type DeepFlat, type DeepFlatSync };
                ~~~~~~~~

node_modules/@fxts/core/dist/types/types/DeepFlat.d.ts:4:30 - error TS1005: ',' expected.

4 export { type DeepFlat, type DeepFlatSync };
```


but I don't know where code build faild  
Then there is a suspicion.

```ts
{
  const emojiSet = new Set(
    await toArray(map((a: Algorithm) => a.idx, isClickedByUser)),
  );
  const reportSet = new Set(
    await toArray(
      map((a: ReportAlgorithm) => a.algorithm.idx, reportAlgorithm),
    ),
  );

  return pipe(
    algorithmList,
    filter((alg) => !reportSet.has(alg.idx)),
    map((alg) => ({
      ...alg,
      emojiCount: alg.emojis.length,
      isClicked: emojiSet.has(alg.idx),
    })),
    toArray,
  );
};
```

### 🙁 Actual behavior

build faild!

### 🙂 Expected behavior

build success!

### Version Information

<!-- Please let me know your environment -->

- Operating system: Window11
- Typescript: 4.4.4
- Nodejs: v16.13.2
- My fxts version : 0.5.0

--- a/src/Lazy/flat.ts
+++ b/src/Lazy/flat.ts
@@ -1,7 +1,7 @@
 import concurrent, { isConcurrent } from "./concurrent";
 import last from "../last";
 import IterableInfer from "../types/IterableInfer";
-import { type DeepFlat, DeepFlatSync } from "../types/DeepFlat";
+import type { DeepFlat, DeepFlatSync } from "../types/DeepFlat";
 import { empty, isAsyncIterable, isIterable } from "../_internal/utils";
 import { Reject, Resolve } from "../types/Utils";
 import append from "./append";

--- a/src/Lazy/flatMap.ts
+++ b/src/Lazy/flatMap.ts
@@ -3,7 +3,7 @@ import flat from "./flat";
 import { isAsyncIterable, isIterable } from "../_internal/utils";
 import Awaited from "../types/Awaited";
 import IterableInfer from "../types/IterableInfer";
-import { type DeepFlat, type DeepFlatSync } from "../types/DeepFlat";
+import type { DeepFlat, DeepFlatSync } from "../types/DeepFlat";
 
 type ReturnFlatMapType<
   A extends Iterable<unknown> | AsyncIterable<unknown>,

--- a/src/types/DeepFlat.ts
+++ b/src/types/DeepFlat.ts
@@ -14,4 +14,4 @@ type DeepFlatSync<A, B extends number = 0> = A extends Iterable<infer U>
     : DeepFlat<U, Sub<B, 1>>
   : A;
 
-export { type DeepFlat, type DeepFlatSync };
+export type { DeepFlat, DeepFlatSync };

